sys/posix/pthread/Makefile

```diff
@@ -1 +1,6 @@
+# exclude pthread_tls if not selected explicitly
+ifeq (,$(filter pthread_tls,$(USEMODULE)))
+  SRC := $(filter-out pthread_tls.c,$(wildcard *.c))
```

Comment:
I do not really like this pattern with the filter-out, I think it would be clearer to just create a subdirectory with a normal module.

Reply:
Or use submodules?

Reply:
Need to rename the file pthread_tls.c -> tls.c or make the submodule name pthread_pthread_tls